### PR TITLE
Fix some errors. Clarify undefined things.

### LINKED ISSUE #217: Should `G` or `MacError` be clarified?
As far as I see, `MacError` is not properly defined. While I understand what it refers to, it might be good to define it for someone that might get confused.

On line 679, also, on the `DeriveAuthKeyPair`:

```
2. public_key = private_key * G
```
the value `G` is not defined elsewhere, which might be confusing for a casual reader.. should it be better defined?

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #217.

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -1545,10 +1552,13 @@ Input:
 Output:
 - session_key, the shared session secret if, and only if, KE3 is valid, nil otherwise.
```
Comment (chris-wood) [inline draft-irtf-cfrg-opaque.md:1553]:
```suggestion
- session_key, the shared session secret if and only if KE3 is valid.
```

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -1545,10 +1552,13 @@ Input:
...
+Exceptions:
+- HandshakeError, when the handshake fails
+
 Steps:
-1. if ct_equal(ke3.client_mac, state.expected_client_mac):
-2.    Output state.session_key
-3. Output nil
+1. if !ct_equal(ke3.client_mac, state.expected_client_mac):
+2.    raise HandshakeError
+3. Output state.session_key
```
Comment (chris-wood) [inline draft-irtf-cfrg-opaque.md:1561]: This change makes the client and server interfaces different: both either return something, or toss an exception.

**Review by kevinlewi — APPROVED** [review]
Great changes!

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -676,7 +680,7 @@ Output:
 
 Steps:
 1. private_key = HashToScalar(seed, dst="OPAQUE-HashToScalar")
-2. public_key = private_key * G
+2. public_key = ScalarBaseMult(private_key)
```
Comment (bytemare) [inline draft-irtf-cfrg-opaque.md:683]: `ScalarBaseMult` is not defined in this document, if I'm not mistaken

Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:683]: It's in the VOPRF document, Section 2.1 (see the citation added in this PR, which also covers where HashToScalar is introduced.)

**Pull request merged**